server/graphql/v2/query/RequiredBankInformationQuery.ts

```diff
@@ -0,0 +1,24 @@
+import { GraphQLNonNull, GraphQLString } from 'graphql';
+
+import RequiredBankInformation from '../object/RequiredBankInformation';
+import models from '../../../models';
+
+const RequiredBankInformationQuery = {
+  type: RequiredBankInformation,
+  args: {
+    slug: {
+      type: new GraphQLNonNull(GraphQLString),
+      description: 'The public id identifying the host collective',
+    },
+  },
+  async resolve(_, args, req): Promise<any> {
+    if (!req.remoteUser) {
+      return null;
+    }
+    const slug = args.slug.toLowerCase();
+    req.collective = await models.Collective.findBySlug(slug);
```

Comment:
We should avoid using `context` like this whenever possible because context is global (it is shared with by all resolvers). 

1. We're going to have problems if a different endpoint starts using the same `collective` field for a totally different purpose.
2. If you're not aware of both files, it won't be clear why we're touching the context (where does this `collective` comes from?).

An easy way to solve this is to have a `CollectiveId` field inside of `RequiredBankInformation` and return `{ CollectiveId }` instead of an empty object. Better: I'd suggest cherrypicking what @znarf did in https://github.com/opencollective/opencollective-api/blob/972431161bbc79d8531445e414a7333925e5321f/server/graphql/v2/input/AccountInput.js so that you take an `account` in args rather than a `slug` or an `id`.

Reply:
Perfect, thanks!